### PR TITLE
Create target and topic models

### PR DESCRIPTION
**Pull request by leandrodzp**
Description
* Models and tests for Target and Topic created

### CODE REVIEW

File: app/views/api/v1/targets/create.json.jbuilder [hunk 1]
```diff
@@ -0,0 +1,8 @@
+json.target do
+  json.id           @target.id
```
Comment (gonzaM) [inline app/views/api/v1/targets/create.json.jbuilder]: You can use [extract!](https://www.rubydoc.info/github/rails/jbuilder/Jbuilder:extract!) here.
For the topic name, you can use delegate in the target model.

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,14 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+topics = [
```
Comment (gonzaM) [inline db/seeds.rb]: Use `%w` to declare the array.

File: spec/requests/api/v1/targets/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+describe 'POST api/v1/targets', type: :request do
+  let(:user) { create(:user, confirmed_at: Time.zone.now) }
```
Comment (gonzaM) [inline spec/requests/api/v1/targets/create_spec.rb]: Align the lets

File: db/migrate/20180910173758_create_targets.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class CreateTargets < ActiveRecord::Migration[5.1]
+  def change
+    create_table :targets do |t|
+      t.references :topic, foreign_key: true
+      t.references :user, foreign_key: true
+      t.string :title
+      t.integer :area_length
```
Comment (letiesperon) [inline db/migrate/20180910173758_create_targets.rb]: What is area length? 🤔  You mean radius?

File: db/migrate/20180910173758_create_targets.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+class CreateTargets < ActiveRecord::Migration[5.1]
+  def change
+    create_table :targets do |t|
+      t.references :topic, foreign_key: true
+      t.references :user, foreign_key: true
+      t.string :title
```
Comment (letiesperon) [inline db/migrate/20180910173758_create_targets.rb]: Maybe you want to add `null: false` restrictions, and indexes to make the search faster

File: app/models/user.rb [hunk 6]
```diff
@@ -41,6 +41,8 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable, :validatable, :trackable,
          :recoverable, :rememberable
 
+  has_many :targets
```
Comment (letiesperon) [inline app/models/user.rb]: Maybe you want `dependent: destroy` so when the user is deleted so are his targets

File: app/models/target.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+class Target < ApplicationRecord
+  belongs_to :user
+  belongs_to :topic
+
+  validates :title, presence: true
+  validates :area_length, numericality: { greater_than: 0 }
+  validates :topic, uniqueness: {
+    scope: %i[latitude longitude user],
+    message: "Can't create 2 targets in the same location with the same topic."
```
Comment (letiesperon) [inline app/models/target.rb]: I think we should use internalization already @gonzaM

Comment (gonzaM) [reply app/models/target.rb]: I agree 👍

File: db/migrate/20180910173758_create_targets.rb (hunk 4 above)
Comment (leandrodzp) [reply db/migrate/20180910173758_create_targets.rb]: I get it from the design :( 
screen shot 2018-09-14 at 12 42 08 pm

Comment (letiesperon) [reply db/migrate/20180910173758_create_targets.rb]: Then the area is a square 🤔

File: spec/models/topic_spec.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+#  id         :bigint(8)        not null, primary key
+#  name       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+require 'rails_helper'
+
+describe Topic do
+  describe 'validations', type: :model do
```
Comment (mdornel) [inline spec/models/topic_spec.rb]: The type of test should be present only once, at the beginning of the test, not in every describe block. Check this example https://relishapp.com/rspec/rspec-rails/docs/model-specs

File: app/models/target.rb [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  }
+  validate :less_than_10_targets
+
+  private
+
+  def less_than_10_targets
```
Comment (mdornel) [inline app/models/target.rb]: What if the allowed target quantity changes in the future? Say 5, or 15 or whateaver.
You will have to change this method name. I think that you can name this validation better.

**Comment by leandrodzp** [comment]
Done :)

**Comment by leandrodzp** [comment]
done

File: spec/models/topic_spec.rb [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
```
Comment (mdornel) [inline spec/models/topic_spec.rb]: Move `type: :model` to the top of the test, should look like this 
```
describe Topic, type: :model do
...
end
```

This way you won't have to include it for new describe blocks in your test

File: spec/requests/api/v1/targets/create_spec.rb [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+        create(
+          :target,
+          user: user,
+          latitude: params[:target][:latitude],
+          longitude: params[:target][:longitude],
+          topic_id: params[:target][:topic_id]
+        )
+      end
+
+      it 'returns error' do
```
Comment (mdornel) [inline spec/requests/api/v1/targets/create_spec.rb]: Besides checking the http status on a failed request, consider to add an expectation for the error message.

**Comment by leandrodzp** [comment]
Done :)

**Pull request merged**